### PR TITLE
Some tweaks

### PR DESCRIPTION
**Pull request by bahamat**
* set hostname based on `sdc:hostname` metadata (if any)
* get dataset mount point instead of assuming

### CODE REVIEW

File: src/actions.rs [hunk 1]
```diff
@@ -56,7 +56,17 @@ pub fn create_dataset<T: AsRef<str>>(dataset: T) -> Result<PathBuf> {
...
+    mp_cmd.env_clear();
+    mp_cmd.args(&["get","-Ho","value","mountpoint", dataset]);
+    let mp = mp_cmd.output()?;
+    if !mp.status.success() {
+        let err = String::from_utf8_lossy(&mp.stderr);
+        bail!("Unable to determine dataset {} mountpoint: {}", &dataset, err);
+    }
+    let mountpoint = String::from_utf8_lossy(&mp.stdout).trim().to_string();
+
+    let zroot: PathBuf = [&mountpoint, "root"].iter().collect();
```
Comment (papertigers) [inline src/actions.rs]: Nit: You could make the from_utf8 stuff hard error and avoid the extra to_string().

```rust
+     let mountpoint = String::from_utf8(mp.stdout).context("invalid utf8 found in dataset mountpoint")?;
+
+     let zroot: PathBuf = [mountpoint.trim(), "root"].iter().collect();
```

**Comment by papertigers** [comment]
Minor nit.
Up to you which version you like better.
Otherwise I am happy with this. 

Tested in the global zone of an OmniOS box.

File: src/actions.rs (hunk 1 above)
Comment (bahamat) [reply src/actions.rs]: Ok, yes I like that better.

Comment (bahamat) [reply src/actions.rs]: Fixed.

**Review by papertigers — APPROVED** [review]

**Pull request merged**